glusterd2/commands/volumes/volume-create.go

```diff
@@ -73,6 +73,25 @@ func createVolinfo(req *api.VolCreateReq) (*volume.Volinfo, error) {
 		v.Type = volume.DistReplicate
 	}
 
+	if req.DisperseData != 0 || req.DisperseRedundancy != 0 {
```

Comment:
As I highlighted in another PR of adding arbiter support, my suggestion would be to have all the integration feature specific validations to be captured in a function so that we don't heavyload the GD2 core management functions with too many lines of changes.